[plugin.audio.mixcloud] 3.0.0

jackyNIX opened the pull request:
### Description
      [new] python 3 migration
      [new] new logo, icons and other art
### Checklist:
- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

basrieter commented on plugin.audio.mixcloud/lib/base.py:
```diff
@@ -0,0 +1,283 @@
+    def execute(self):
+        Utils.log('BaseBuilder.execute()')
+        ret = self.build()
+        if ret != None:
```
Comparison with `None` should be done with `is`: `if ret is None`

basrieter commented on plugin.audio.mixcloud/lib/base.py:
```diff
@@ -0,0 +1,283 @@
+        if isinstance(nextOffset, list):
+            nextOffsetEx = nextOffset[1]
+            nextOffset = nextOffset[0]
+        if (nextOffset and (nextOffset > 0)) or ((nextOffsetEx != None) and (nextOffsetEx > 0)):
```
Comparison with `None` should be done with `is` or `is not`

basrieter commented on plugin.audio.mixcloud/lib/base.py:
```diff
@@ -0,0 +1,283 @@
+            nextOffset = nextOffset[0]
+        if (nextOffset and (nextOffset > 0)) or ((nextOffsetEx != None) and (nextOffsetEx > 0)):
+            parameters = {'mode' : self.mode, 'key' : self.key, 'offset' : nextOffset}
+            if nextOffsetEx != None:
```
Comparison with `None` should be done with `is` or `is not`

basrieter commented on plugin.audio.mixcloud/addon.xml:
```diff
@@ -0,0 +1,42 @@
+      [new] new logo, icons and other art
+    </news>
+    <platform>all</platform>
+    <language></language>
```
Please fill this tag

basrieter commented on plugin.audio.mixcloud/lib/listbuilder.py:
```diff
@@ -0,0 +1,289 @@
+        cloudcasts = MixcloudInterface().getList(self.key, {'offset' : self.offset})
+        for cloudcast in cloudcasts.items:
+            # user aborted
+            if xbmc.Monitor().abortRequested():
```
It would be better to have a single `xbmc.Monitor()` instance here, and call `abortRequested()` on that single instance, instead of creating a lot of new `xbmc.Monitor()` instances.

basrieter commented on plugin.audio.mixcloud/lib/listbuilder.py:
```diff
@@ -0,0 +1,289 @@
+            index = 0
+            for keyitem in searchHistory.data:
+                # user aborted
+                if xbmc.Monitor().abortRequested():
```
Same here with the Monitor

basrieter commented on plugin.audio.mixcloud/lib/mixcloud.py:
```diff
@@ -0,0 +1,275 @@
+                data = response['data']
+                for item in data:
+                    # user aborted
+                    if xbmc.Monitor().abortRequested():
```
And here again with the Monitor()

basrieter commented on plugin.audio.mixcloud/lib/mixcloud.py:
```diff
@@ -0,0 +1,275 @@
+                offset = parameters['offset']
+            for keyitem in keylist:
+                # user aborted
+                if xbmc.Monitor().abortRequested():
```
And here......

basrieter commented on plugin.audio.mixcloud/lib/mixcloud.py:
```diff
@@ -0,0 +1,275 @@
+            oathCode = Utils.getSetting('oath_code')
+            while ask:
+                # user aborted
+                if xbmc.Monitor().abortRequested():
```
And here

jackyNIX commented:
change requests: done
commit squashed

basrieter commented on plugin.audio.mixcloud/lib/history.py:
```diff
@@ -0,0 +1,127 @@
+    def writeFile(self):
+        filepath = xbmc.translatePath(__addon__.getAddonInfo('profile')) + self.name + '.json'
+        try:
+            text_file = open(filepath, 'w+')
```
A small remark here. Better do this:
```python
with open(filepath, 'w+') as text_file :
    text_file.write(json.dumps(self.data, indent = 4 * ' '))
```
No closed need then. Now if an exception occurs, the file is not closed.

basrieter commented on plugin.audio.mixcloud/lib/history.py:
```diff
@@ -0,0 +1,127 @@
+        try:
+            # read file
+            if os.path.exists(filepath):
+                text_file = open(filepath, 'r')
```
A small remark here. Better do this:
```python
with open(filepath, 'r') as text_file:
    self.data = json.loads(text_file.read())
```
No closed need then. Now if an exception occurs, the file is not closed.

enen92 commented on plugin.audio.mixcloud/addon.xml:
```diff
@@ -0,0 +1,42 @@
+    <assets>
+      <icon>resources/icon.png</icon>
+      <fanart>resources/fanart.jpg</fanart>
+      <clearlogo>resources/clearlogo.png</clearlogo>
```
we should wait on https://github.com/xbmc/addon-check/pull/212 to be finished, merged and part of a new addon-checker release before considering merging this addon

basrieter replied:
@enen92  the clearlogo is removed. So I am fine with this, unless you think we should still wait?

enen92 commented on plugin.audio.mixcloud/addon.xml:
```diff
@@ -0,0 +1,42 @@
+    </news>
+    <platform>all</platform>
+    <language>en</language>
+    <license>GPL-3.0</license>
```
this identifier does not exist

jackyNIX commented:
change requests done:

- open() changed to with open() as ... and removed the close()
- GPL-3.0 identifier changed to GPL-3.0-or-later
- removed optional artwork (clearlogo and screenshots). Rest of the artwork (icon, fanart and 9 plugin navigation icons) is 140KB together.
